zingolib/src/wallet/keys/unified.rs

```diff
@@ -84,53 +116,82 @@ fn read_write_receiver_selections() {
     }
 }
 
-impl UnifiedSpendCapability {
+impl WalletCapability {
     pub fn addresses(&self) -> &[UnifiedAddress] {
         &self.addresses
     }
 
-    pub fn transparent_child_keys(&self) -> &Vec<(usize, secp256k1::SecretKey)> {
-        &self.transparent_child_keys
+    pub fn transparent_child_keys(&self) -> Result<&Vec<(usize, secp256k1::SecretKey)>, String> {
+        if self.transparent.can_spend() {
+            Ok(&self.transparent_child_keys)
+        } else {
+            Err("Wallet is not capable to spend transparent funds.".to_string())
```

Comment:
"The wallet is not capable of spending transparent funds."

Reply:
a84c39daf77360b612a7bd532058753248b77793